server/src/api/user.rs

```diff
@@ -115,14 +124,25 @@ mod handlers {
         status_response(StatusCode::NO_CONTENT)
     }
 
-    /// Redeems an invitation. An invitation includes a WireGuard keypair generated by either the server
-    /// or a peer with admin rights.
-    ///
-    /// Redemption is the process of an invitee generating their own keypair and exchanging their temporary
-    /// key with their permanent one.
-    ///
-    /// Until this API endpoint is called, the invited peer will not show up to other peers, and once
-    /// it is called and succeeds, it cannot be called again.
+    /// Report any other endpoint candidates that can be tried by peers to connect.
+    pub async fn candidates(
+        contents: Vec<Endpoint>,
+        session: Session,
+    ) -> Result<Response<Body>, ServerError> {
+        let conn = session.context.db.lock();
+        let mut selected_peer = DatabasePeer::get(&conn, session.peer.id)?;
+        selected_peer.update(
+            &conn,
+            PeerContents {
+                candidates: contents,
+                ..selected_peer.contents.clone()
+            },
```

Comment:
May it happen that the client sends its "main peer address" that we already have as one of their candidates? Shall we filter it out? Shall we check in client and/or in server that candidates are unique?

Reply:
Yeah, that could certainly happen and could waste a 5 second step interval if it couldn't connect the first time. I'll add deduping and also just a total limit to the number of candidates.